### PR TITLE
[deps] Update to vLLM 0.26.0

### LINKED ISSUE #64: Update to plugin to vLLM v0.26
## Summary
Compatibility check for vLLM `v0.25.1 → v0.26.0` across every API the plugin couples to (8 subclassed base classes, ~10 constructed dataclasses, 5 monkeypatch targets). The coupling surface is almost entirely stable: one breaking signature change, one behavioral return-type change, and a few new optional fields; everything else is unchanged or byte-identical. Release highlights (GPU/ROCm/XPU kernels, new models, quantization, KV-offloading) don't touch the TT backend.

## Diffs relevant to the plugin (0.25.1 -> 0.26.0)

**Breaking**
- `v1/worker/block_table.py` · `MultiGroupBlockTable.__init__`: `max_model_len` removed;
  `max_num_blocks` now required (was optional/auto-derived); new optional `slot_mapping_modes`.

**Behavioral**
- `v1/core/sched/interface.py` · `SchedulerInterface.finish_requests`: returns `list[Request]` (was `list[tuple[str, int]]`); `EngineCoreProc._send_abort_outputs` now consumes `list[Request]`.
- `v1/core/sched/output.py` · `SchedulerOutput`: new optional fields `kv_cache_block_copies`, `scheduled_encoder_input_stats`.
- `v1/metrics/stats.py` · `SchedulerStats`: new optional `iteration_details`.

**Dependencies (`requirements/common.txt`)**
- `diskcache` removed; `llguidance` platform marker gains `s390x`. No other changes.

**Unchanged / byte-identical (no plugin-relevant diff)**
Worker/Platform/Executor (`worker_base.py`, `executor/abstract.py` byte-identical; `platforms/interface.py` doc/log-only), `Scheduler`/`AsyncScheduler` (+ 17-method `SchedulerInterface` abstract set, `create_request_queue`, `EngineCoreOutputs`, `GrammarOutput`), model-runner outputs/sampling/logits (`ModelRunnerOutput`, `AsyncModelRunnerOutput`, `SamplingMetadata`, `Sampler`, `CachedRequestState`), config/KV-cache/loader/registry/tasks/multimodal, all 5 `platform.py` monkeypatch targets, and the 7 `vllm.utils.*` helpers. Launcher (`CoreEngineLauncher`/ `EngineLaunchPlan`) still absent.

## Method
Diffed installed `0.25.1+empty` source against upstream `vllm-project/vllm@v0.26.0`, per coupled symbol.

### PR DESCRIPTION
**Pull request by sanjaradylov**
## TL;DR

Builds the plugin against **vLLM 0.26.0** (from 0.25.1).

## Details

**Code**
- `src/vllm_tt_plugin/input_batch.py` — 0.26.0's `MultiGroupBlockTable.__init__` drops `max_model_len` and makes `max_num_blocks` required (adds optional `slot_mapping_modes`). Pass the per-group block budget explicitly (`max_num_blocks=[cdiv(max_model_len, bs) for bs in block_sizes]`, + `cdiv` import). TT runs `cp_world_size == 1` and 0.26.0 still applies the internal 128-alignment, so behavior is unchanged.
- `tests/test_block_scheduler.py` — 0.26.0 changes `finish_requests` to return `list[Request]` (was `list[tuple[str, int]]`). The engine-reset abort path forwards this list unchanged (upstream's `_send_abort_outputs` now also expects `list[Request]`); the stale assertion is updated to match.

**Version bump**
- `docs/install-vllm-tt.sh` — pinned source build + `requirements/common.txt` URL → `v0.26.0`.
- `README.md`, `docs/diffusion-gemma.md` — version references.

## Related Artifacts

Resolves #64 

## Validation

- Built `vllm 0.26.0+empty` from source; **329/329** host-only tests pass (`pytest tests/ --ignore=tests/tt`).
- Actions [passing](https://github.com/tenstorrent/tt-metal/actions/runs/32199971504).

## Checklist

- [x] This PR is focused on a single logical change.
- [x] I added or updated tests where applicable.
- [x] I ran the relevant checks such as unit tests and Actions and recorded them above.
- [x] I updated documentation where applicable.

### CODE REVIEW

**Comment by sanjaradylov** [comment]
Looks like all tests are passing.
If there are other, more comprehensive tests (e.g., performance evaluations) scheduled by the users of the plugin, I'd be happy to get feedbacks.
Otherwise, proposing as a ready feature.

**IMPORTANT**: running this version with vLLM v0.25.* will definitely result in runtime errors tracing back to the input processing.

CC @viktorpusTT

File: README.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ for the appropriate tt-metal and vLLM commits.
 vLLM requires Python `>=3.10,<3.14`. Python 3.10.12 is the default `python3` on
 Ubuntu 22.04.
 
-The installation script builds vLLM `0.25.1` from source with
+The installation script builds vLLM `0.26.0` from source with
 `VLLM_TARGET_DEVICE=empty`. Other vLLM versions are not tested.
 
 To install against vLLM `0.24.0` instead, check out the `compat/vllm-0.24.0`
```
Comment (viktorpusTT) [inline README.md]: Turn this into a table? Or state a generic rule with an example?

**Review by viktorpusTT — APPROVED** [review]
Thank you. Please create `compat/vllm-0.25.1` tag.

**Comment by viktorpusTT** [comment]
One more finding:
TTLaneCoordinator.finish_requests still returns None. This is the plugin’s only direct SchedulerInterface implementor. 0.26.0’s interface returns list[Request]. EngineCoreProc.pause_scheduler(mode="abort") and shutdown-abort both do aborted_reqs = self.scheduler.finish_requests(None, ...) then _send_abort_outputs(aborted_reqs), which reads request.client_index. The coordinator still annotates -> None, omits | None on request_ids, and discards each lane’s return. _send_abort_outputs(None) is falsy, so lane-DP abort notifications are dropped while the requests have already been finished. TTScheduler inherits the new contract and the PR only updated that test. The hole existed on 0.25.1 too; this bump is the contract change the PR is adapting to, and it missed the subtype that does not inherit. Suggestion: Match the 0.26.0 signature, concatenate per-lane list[Request], return it. Add a coordinator test that finish_requests(None, ...) returns the aborted Request objects.
